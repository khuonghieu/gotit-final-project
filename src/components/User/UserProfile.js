import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from '@gotitinc/design-system';

function UserProfile({ currentUser }) {
  return (

    <div style={{ width: 350 }}>
      <span className="u-marginRightSmall">
        <Badge variant="positive_subtle">{currentUser ? currentUser.name : 'Not logged in'}</Badge>
      </span>
    </div>

  );
}

UserProfile.propTypes = {
  currentUser: PropTypes.object,
};

export default UserProfile;
