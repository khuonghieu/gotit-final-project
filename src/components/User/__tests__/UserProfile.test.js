import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createBrowserHistory } from 'history';
import { UserProfile } from '../UserProfile';

configure({ adapter: new Adapter() });

describe('components/User/UserProfile.js', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      user: {},
    };
  });
  const setup = () => {
    wrapper = shallow(
      <UserProfile {...props} />,
    );
  };

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should display "not logged in" if currentUser is null', () => {
    setup();
    expect(wrapper.text()).toBe('Not logged in');
  });

  it('should display current user name', () => {
    props = {
      ...props,
      user: {
        loggedIn: true,
        currentUser: {
          id: 1,
          username: 'user',
          email: 'user@gotitapp.co',
          name: 'testuser',
          created: '2015-08-05T08:40:51.620Z',
          updated: '2018-04-03T08:40:51.620Z',
        },
      },

    };
    setup();
    expect(wrapper.text()).toBe('testuser');
  });
});
