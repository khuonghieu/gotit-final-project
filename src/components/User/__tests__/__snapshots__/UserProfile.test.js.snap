// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/User/UserProfile.js should render correctly 1`] = `ShallowWrapper {}`;
